spec.bs

```diff
@@ -1462,86 +1480,88 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
+        1. [=list/For each=] |ig| of |groups|:
+          1. If |ig|'s [=interest group/bidding url=] is null, [=iteration/continue=].
           1. If |perBuyerCumulativeTimeout| is not null and is less than |perBuyerTimeout|, then set
             |perBuyerTimeout| to |perBuyerCumulativeTimeout|.
           1. Let |generateBidStartTime| be |settings|'s
             [=environment settings object/current monotonic time=].
-          1. Set |generatedBid| to the result of [=generate a bid=] given
-            |allTrustedBiddingSignals|, |auctionSignals|, a [=map/clone=] of |browserSignals|,
-            |perBuyerSignals|, |directFromSellerSignalsForBuyer|, |perBuyerTimeout|, |expectedCurrency|,
-            and |ig|.
-          1. Set |ig|'s [=interest group/ads=] to |originalAds|.
-          1. Set |ig|'s [=interest group/ad components=] to |originalAdComponents|.
-          1. Let |generateBidDuration| be the [=duration from=] |generateBidStartTime| to
-            |settings|'s [=environment settings object/current monotonic time=], in milliseconds.
-          1. If |perBuyerCumulativeTimeout| is not null, then decrement |perBuyerCumulativeTimeout|
-            by |generateBidDuration|.
+          1. Let |generatedBid| be the result of [=generate a bid=] given |allTrustedBiddingSignals|,
+            |auctionSignals|, a [=map/clone=] of |browserSignals|, |perBuyerSignals|,
+            |perBuyerTimeout|, |expectedCurrency|, |ig|, and |auctionStartTime|.
+          1. Let |generateBidDuration| be the [=duration from=] |generateBidStartTime| to |settings|'s
+            [=environment settings object/current monotonic time=], in milliseconds.
+          1. If |perBuyerCumulativeTimeout| is not null, decrement |perBuyerCumulativeTimeout| by
+            |generateBidDuration|.
           1. If |generatedBid| is failure, [=iteration/continue=].
-        1. [=list/Insert=] |generatedBid|'s [=generated bid/interest group=] in |bidIgs|.
-        1. [=Score and rank a bid=] with |auctionConfig|, |generatedBid|, |leadingBidInfo|,
-          |decisionLogicScript|, |directFromSellerSignalsForSeller|, |dataVersion|, |auctionLevel|,
-          |componentAuctionExpectedCurrency|, and |topLevelOrigin|.
+          1. If [=query generated bid k-anonymity count=] given |generatedBid| returns false:
+  
+            Note: [=Generate a bid=] is now rerun with only k-anonymous [=interest group/ads=] to give
+            the buyer a chance to [=generate a bid=] for k-anonymous [=interest group/ads=]. Allowing
+            the buyer to first [=generate a bid=] for non-k-anonymous [=interest group/ads=] provides a
+            mechanism to bootstrap the k-anonymity count, otherwise no [=interest group/ads=] would
+            ever trigger [=increment k-anonymity count=] and all ads would fail
+            [=query k-anonymity count=].
+            1. TODO: Run [=score and rank a bid=] on |generatedBid| to find the highest scoring bid

```

Comment:
Could you make this either an `Issue: ` so that the TODO is more pronounced (and accumulates at the bottom of the spec with the rest of the "Issues", or an XXX box?

`Issue:` is probably preferred, and maybe a follow-up bug describing what has to be done. It's not very clear what remaining work there is here.

Reply:
oh I think there's some indent problems here during rebase (just realized that I had "ignore whitespace" selected, so these blocks didn't show up as difference). This PR shouldn't be touching these. Let me see what's going on here.

Reply:
ah I see, this PR added another for-loop layer so that the whole block gets indented once more. Since this is technically un-related to this PR, I'll do this in a separate PR as well.

Reply:
Oh, it's just diff being weird, aligning the wrong lines.  Nevermind.

Reply:
no, your PR didn't add |auctionStartTime|, it's just github differing wrong lines. Your line 1512 should be compared to the [old line 1492](https://github.com/WICG/turtledove/pull/954/files?w=0#diff-6f5a1d8263b0b0c42e2716ba5750e3652e359532647ac934c1c70086ae3ceddaL1492 ), then it's just indent change. And it does find a bug in the original spec that [old line 1530](https://github.com/WICG/turtledove/pull/954/files?w=0#diff-6f5a1d8263b0b0c42e2716ba5750e3652e359532647ac934c1c70086ae3ceddaL1530) (which your line 1512 is comparing to) needed an |auctionStartTime| which should have been added. I'll fix this as well separately

Reply:
So is the indent issue "solved" here then? Can this thread be "resolved" accordingly?

Reply:
Yes, I copied the diff to a diff tool and made the new change indent left, and they were the same (with a few small expected changes related to this PR)